lib/smart_proxy_remote_execution_ssh/multiplexed_ssh_connection.rb

```diff
@@ -119,10 +127,10 @@ def try_auth_method(method)
       if pm.status.zero?
         logger.debug("Established connection using authentication method #{method.name}")
         @socket = socket_file
-        true
+        [true, pm]
```

Comment:
I know, we all rather write in Go :D

But looking at this method, it looks weird (from Ruby perspective) in both cases: old and new.

What if we make `try_auth_method` always return `pm`? `true/false` seems to not be helpful anyway. Especially when the name of the method is without `?`.